### PR TITLE
Learn Sass page and code example helper.

### PR DESCRIPTION
**Pull request by sanajaved7**

### CODE REVIEW

**Comment by sanajaved7** [comment]
[port learn sass page](https://trello.com/c/Ih0g0QTR/27-port-learn-sass-page)

File: source/guide.md [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+With that in mind, here's an example of some typical styles for a site's
+navigation:
+
+{% capture example_id %}
+{{ example_id | plus: 1 }}
+{% endcapture %}
+{% capture ui_id %}
+{{ ui_id | plus: 3 }}
+{% endcapture %}
```
Comment (mirisuzanne) [inline source/guide.md]: I don't know all the minutia of liquid, but would it work a bit more compact like this?

```suggestion
{% assign example_id = example_id | plus: 1 %}
{% assign ui_id = ui_id | plus: 3 %}
```

Does ui_id always increase by three? Would it make sense to create a filter that hides the magic number, and is instead like `| incrementId` and `| incrementUiId`?

File: source/_layouts/base.liquid [hunk 2]
```diff
@@ -43,6 +43,7 @@
 
       gtag('config', 'UA-535380-14');
     </script>
+    <script src="/assets/dist/js/sass.js"></script>
```
Comment (jgerigmeyer) [inline source/_layouts/base.liquid]: I'm curious why this change was needed? I'd lean toward deferring the JS unless it's really needed before content is rendered...

File: source/guide.md [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+[Partials](#topic-4)
+[Modules](#topic-5)
+[Mixins](#topic-6)
+[Inheritance](#topic-7)
+[Operators](#topic-8)
+[Lists](#topic-9)
+
+</nav> -->
+
+## Preprocessing
```
Comment (jgerigmeyer) [inline source/guide.md]: Looks like each of these should be in a unique `<section>` with an ID.

File: source/guide.md [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+[Variables](#topic-2)
+[Nesting](#topic-3)
+[Partials](#topic-4)
+[Modules](#topic-5)
+[Mixins](#topic-6)
+[Inheritance](#topic-7)
+[Operators](#topic-8)
+[Lists](#topic-9)
+
+</nav> -->
```
Comment (jgerigmeyer) [inline source/guide.md]: It looks like this `navigation` block isn't being used in production unless we use the `has_navigation` or `has_both_sidebars` layout. Might be worth considering that, but a low priority.

File: source/assets/js/vendor/index.ts [hunk 5]
```diff
@@ -1,4 +1,11 @@
-import 'jquery-ui';
+// import jQuery UI "Tabs" widget and dependencies
+// https://jqueryui.com/tabs/
+import 'jquery-ui/ui/version';
+import 'jquery-ui/ui/widget';
+import 'jquery-ui/ui/keycode';
+import 'jquery-ui/ui/unique-id';
+import 'jquery-ui/ui/safe-active-element';
+import 'jquery-ui/ui/widgets/tabs';
```
Comment (jgerigmeyer) [inline source/assets/js/vendor/index.ts:8]: If we don't like this somewhat-opaque list, we could download a single compiled file and just include it. 🤷

File: source/guide.md [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+}
+
+===
+
+$font-stack: Helvetica, sans-serif
+$primary-color: #333
+
+body
+font: 100% $font-stack
+color: $primary-color" | codeExample: true, 'scss' -%}
```
Comment (jgerigmeyer) [inline source/guide.md]: The indentation here is unfortunate and makes it difficult to read. Could we consider a `{% capture %}` block instead?

File: source/guide.md [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+watch flag to your command, like so:
+
+```
+sass --watch input.scss output.css
+```
+
+You can watch and output to directories by using folder paths as your
+input and output, and separating them with a colon. In this example:
+
+{% renderFile 'source/code-snippets/_homepage-sass-watch.md' %}
```
Comment (jgerigmeyer) [inline source/guide.md]: It seems a bit odd that one of these is a partial (only used here), and the other is inline code. 🤷

File: source/guide.md [hunk 8]
```diff
@@ -0,0 +1,441 @@
...
+body
+font: 100% $font-stack
+color: $primary-color" | codeExample: true, 'scss' -%}
+
+{% assign example_id = 1 %}
+{% assign ui_id = 1 %}
+{% render 'code_example',
+code: code,
+example_id: example_id,
+ui_id: ui_id %}
```
Comment (jgerigmeyer) [inline source/guide.md]: Hrm, this is tricky... The liquid file is included here in a Markdown template, where it is then parsed as Markdown. So that means whitespace in the Liquid template is significant, which is causing extra `<p>` tags where we don't want them (specifically when the code sample contains two consecutive newlines `\n\n`). 😖 

I'm struggling to think of a good solution... I think it's a fundamental problem/risk any place we're embedding markup in a markdown file. Maybe our standard approach should shift to Liquid templates, with Markdown embedded selectively, instead of the other way around? Markdown is dangerous when it's used in unexpected places! 😅

File: source/_includes/code_example.liquid [hunk 9]
```diff
@@ -0,0 +1,81 @@
+{% assign first_tab_id = ui_id %}
+{% assign second_tab_id = first_tab_id | plus: 1 %}
+{% assign third_tab_id = second_tab_id | plus: 1 %}
```
Comment (jgerigmeyer) [inline source/_includes/code_example.liquid]: Instead of using incrementing IDs that we have to pass in from the place this is used, what about having this template itself auto-generate a unique ID, and then internally increment it? I'm not sure why the outer template needs to care about this, and I also don't think it's very important to have them easily readable and incremented. Is that a requirement of the jQuery UI widget?

**Review by jgerigmeyer — COMMENTED** [review]
Looks like a great start!

File: source/_layouts/base.liquid (hunk 2 above)
Comment (sanajaved7) [reply source/_layouts/base.liquid]: Originally the tabs functionality wasn't working without this but I checked again just now and its working fine 🙃 will undo the change

File: source/assets/js/vendor/index.ts (hunk 5 above)
Comment (sanajaved7) [reply source/assets/js/vendor/index.ts:8]: This is fine with me 👍🏼

File: source/guide.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+---
+layout: guide
+title: Sass Basics
+introduction: >
+  Before you can use Sass, you need to set it up on your project. If you want to
+  just browse here, go ahead, but we recommend you go install Sass first. [Go
+  here](/install) if you want to learn how to get everything set up.
+---
```
Comment (jgerigmeyer) [inline source/guide.md]: @sanajaved7 Is there a benefit to having this file? I think we could delete this, and move `_layouts/guide.liquid` here instead (`source/guide.liquid`) and it would work the same?

File: source/_layouts/guide.liquid [hunk 11]
```diff
@@ -0,0 +1,481 @@
...
+  Before you can use Sass, you need to set it up on your project. If you want to
+  just browse here, go ahead, but we recommend you go install Sass first. [Go
+  here](/install) if you want to learn how to get everything set up.
+---
+
+<section id="topic-1">
+  {% markdown %}## Preprocessing
+    
+    CSS on its own can be fun, but stylesheets are getting larger, more
+    complex, and harder to maintain. This is where a preprocessor can help.
```
Comment (jgerigmeyer) [inline source/_layouts/guide.liquid]: For readability here, I might suggest we only selectively indent in files like this? Like maybe:

```suggestion
<section id="topic-1">

{% markdown %}

## Preprocessing
    
CSS on its own can be fun, but stylesheets are getting larger, more
complex, and harder to maintain. This is where a preprocessor can help.
```

What do you think?

File: source/_layouts/guide.liquid [hunk 12]
```diff
@@ -0,0 +1,481 @@
...
+layout: has_no_sidebars
+title: Sass Basics
+introduction: >
+  Before you can use Sass, you need to set it up on your project. If you want to
+  just browse here, go ahead, but we recommend you go install Sass first. [Go
+  here](/install) if you want to learn how to get everything set up.
+---
+
+<section id="topic-1">
+  {% markdown %}## Preprocessing
```
Comment (jgerigmeyer) [inline source/_layouts/guide.liquid]: Our custom `{% markdown %}` tag should be equivalent to the built-in `{% renderTemplate %}` tag... For consistency, maybe we should remove ours, and just use e.g. `{% renderTemplate 'liquid,md' %}`?

File: source/_layouts/guide.liquid [hunk 13]
```diff
@@ -0,0 +1,481 @@
+---
```
Comment (sanajaved7) [inline source/_layouts/guide.liquid]: @jgerigmeyer @mirisuzanne I took a pass at switching over to using a liquid template for the learn sass page / code helper. The indentation works for me locally now and in the code its much easier to read too. I also switched over to a capture block for the code example. Let me know what you think about this approach!

File: source/_layouts/guide.liquid [hunk 14]
```diff
@@ -0,0 +1,481 @@
...
+body
+  font: 100% $font-stack
+  color: $primary-color
+{%- endcapture -%}
+
+  {%- assign parsedExample = captureCode | codeExample: true, 'scss' -%}
+
+  {% assign example_id = 1 %}
+  {% assign ui_id = 1 %}
+  {% render 'code_example', code: parsedExample,example_id: example_id, ui_id: ui_id %}
```
Comment (jgerigmeyer) [inline source/_layouts/guide.liquid]: This seems to work great, and we no longer have the extra `<p>` tags! ✨

**Review by jgerigmeyer — COMMENTED** [review]
@sanajaved7 I like this approach! I have a few minor suggestions for readability -- and I think my questions re incrementing IDs is still open -- but this is on a good track 👏

File: source/_layouts/guide.liquid (hunk 11 above)
Comment (sanajaved7) [reply source/_layouts/guide.liquid]: That indentation works for me 👍🏼

File: source/_layouts/guide.liquid (hunk 12 above)
Comment (sanajaved7) [reply source/_layouts/guide.liquid]: @jgerigmeyer would we have to replace each `markdown` tag with the `renderTemplate` one? Or could we use one `renderTemplate` tag at the top of the file and wrap all the contents in that tag? I'm testing it out locally and it looks like the we might have to do the first option which I feel less excited about 😆

Comment (jgerigmeyer) [reply source/_layouts/guide.liquid]: Yeah, probably the latter... But I could go either way on this. On the one hand it seems odd to add our own custom tag, which just duplicates built-in functionality. But on the other hand our tag is slightly less verbose, so 🤷

Comment (sanajaved7) [reply source/_layouts/guide.liquid]: @jgerigmeyer I got the `renderTemplate` tag working at the top of the file but then decided to keep the `markdown` tag after all because we started to get weird spacing issues again

Comment (jgerigmeyer) [reply source/_layouts/guide.liquid]: Oh, interesting! You got spacing issues replacing `{% markdown %}` and `{% endmarkdown %}` with `{% renderTemplate 'liquid,md' %}` and `{% endrenderTemplate %}` throughout? Or when trying to just use one `{% renderTemplate 'liquid,md' %}` at the top of the file?

The latter won't work, because then everything in the file will be parsed as Markdown -- which is the same problem we had before using a `.md` file. We need to explicitly be able to say when we want Markdown parsing to start and stop, which I think we can do with either `{% renderTemplate 'liquid,md' %}` or `{% markdown %}` equally?

Comment (sanajaved7) [reply source/_layouts/guide.liquid]: @jgerigmeyer yep, I was trying to see if I could get rid of all the various `markdown` tags with one `renderTemplate` tag at the top of the file but it brought back the spacing issues we were seeing with the `.md` file.

Comment (sanajaved7) [reply source/_layouts/guide.liquid]: Looking at both options, I like the `markdown` tag as opposed to `renderTemplate` because it reads a bit more clearly to me so I'm in favor of keeping ours ✨

File: source/guide.liquid [hunk 15]
```diff
@@ -0,0 +1,494 @@
...
+body
+  font: 100% $font-stack
+  color: $primary-color
+{%- endcapture -%}
+
+{%- assign parsedExample = captureCode | codeExample: true, 'scss' -%}
+
+{% assign example_id = 1 %}
+{% assign ui_id = 1 %}
+{% render 'code_example', code: parsedExample,example_id: example_id, ui_id: ui_id %}
```
Comment (jgerigmeyer) [inline source/guide.liquid]: Notes for future discussion after offline discussion with @mirisuzanne:

1. Instead of auto-incrementing IDs, it'd be nice if the IDs/URLs were human-readable, and also don't change on rebuild. What about passing in a custom string ID for each code example?
2. Instead of a filter to generate the data, then passing that data to a LiquidJS template... We could try a shortcode that handles everything -- compiles the Sass, then programmatically passes data to a LiquidJS template (I assume LiquidJS has a node API?), and returns the raw HTML? (Alternatively, the code-example partial itself could accept the input code as an arg, and handle calling the filter.)

File: source/helpers/sass_helpers.ts [hunk 16]
```diff
@@ -0,0 +1,16 @@
+function getImplStatus(status) {
```
Comment (sanajaved7) [inline source/helpers/sass_helpers.ts]: @jgerigmeyer would love some help figuring out how to use this helper method in a new Eleventy filter with types for the `status` arg. I had something like this: `function getImplStatus(status: string | boolean | null){` but then the build command breaks with this error: 
```
[11ty] 1. Error in your Eleventy config file 'eleventy.config.cjs'. (via EleventyConfigError)
[11ty] 2. Unexpected token ':' (via SyntaxError)
[11ty] 
[11ty] Original error stack trace: /Users/sanajaved/Documents/oddbird/sass-site/source/helpers/sass_helpers.ts:1
[11ty] function getImplStatus(status: string | boolean | null){
```
The tsconfig looks like it should include this file since its inside source so I'm not sure what else to try and tinker with 🤔

Comment (jgerigmeyer) [reply source/helpers/sass_helpers.ts]: Should be fixed in https://github.com/oddbird/sass-site/pull/21

File: source/guide.liquid (hunk 15 above)
Comment (jgerigmeyer) [reply source/guide.liquid]: https://www.11ty.dev/docs/languages/liquid/#asynchronous-shortcodes
https://liquidjs.com/api/classes/liquid_.liquid.html#renderFile-1

File: source/guide.liquid [hunk 17]
```diff
@@ -98,12 +98,7 @@ body
   font: 100% $font-stack
   color: $primary-color
 {%- endcapture -%}
-
-{%- assign parsedExample = captureCode | codeExample: true, 'scss' -%}
-{%- assign example_id = 1 -%}
-{%- assign ui_id = 1 -%}
-
-{% render 'code_example', code: parsedExample,example_id: example_id, ui_id: ui_id %}
+{% codeExample variablesCodeExample, 'variables' %}
```
Comment (sanajaved7) [inline source/guide.liquid]: @jgerigmeyer here's my first pass at the `codeExample` short code approach. I used the example name to replace both the numerical example ID number and the ui ID number in links. 

I wanted to also avoid setting this codeExample variable above the short code call but kept getting errors because of the multi-line arg. The short code call worked fine if the variable was something like `{% codeExample "my-test-css", 'variables' %}`. Not sure if you have other workaround suggestions?

Comment (jgerigmeyer) [reply source/guide.liquid]: @sanajaved7 This looks great! I think what you were missing was a [`pairedShortcode`](https://www.11ty.dev/docs/languages/liquid/#paired-shortcode), which allows for contents to be within opening and closing tags instead of the first argument. I adjusted that in https://github.com/oddbird/sass-site/pull/17/commits/59b9d0c9e70fd48961e1eaa9909347663a0e8009 and it seems to work 🎉

File: source/guide.liquid [hunk 18]
```diff
@@ -67,47 +62,57 @@ indented syntax (`.sass`) is more unusual: it uses indentation rather than
...
+  ## Variables
 
-Think of variables as a way to store information that you want to reuse
-throughout your stylesheet. You can store things like colors, font stacks,
-or any CSS value you think you'll want to reuse. Sass uses the `$` symbol
-to make something a variable. Here's an example:
+  Think of variables as a way to store information that you want to reuse
+  throughout your stylesheet. You can store things like colors, font stacks,
+  or any CSS value you think you'll want to reuse. Sass uses the `$` symbol
+  to make something a variable. Here's an example:
```
Comment (mirisuzanne) [inline source/guide.liquid]: Is the indentation here intentional?

**Review by mirisuzanne — APPROVED** [review]
Looks great. I like the change to naming rather than numbering examples. I guess the final step here is using grid for the example tabs? @sanajaved7 is that something that Stacy or I should be starting on? (Or do we want to do that in a separate PR?)

File: source/guide.liquid [hunk 19]
```diff
@@ -0,0 +1,468 @@
...
+indented syntax (`.sass`) is more unusual: it uses indentation rather than
+curly braces to nest statements, and newlines instead of semicolons to
+separate them. All our examples are available in both syntaxes.
+
+{% endmarkdown %}
+
+</div>
+</section>
+
+<hr>
```
Comment (jgerigmeyer) [inline source/guide.liquid:79]: I notice in the existing site these `<hr>` tags are inside the `<section>`, but we have them outside. Probably doesn't matter, and maybe better this way? Just want to be sure it's intentional.

**Comment by sanajaved7** [comment]
@mirisuzanne I'm leaning towards adding the grid as part of this PR personally since the split tabs look a bit funky without it. If we wanted to tackle the work in a separate PR we could just take out the `canSplit` class for now and keep the tabs separate until we're ready.

**Comment by mirisuzanne** [comment]
@sanajaved7 I don't have any reason we should split it out. In that case, it sounds like this is ready for Stacy or I to pick up?

**Comment by jgerigmeyer** [comment]
> @sanajaved7 I don't have any reason we should split it out. In that case, it sounds like this is ready for Stacy or I to pick up?

@mirisuzanne I'm doing a round of review/cleanup on this, so maybe better to wait a bit until I push that?

File: eleventy.config.js [hunk 20]
```diff
@@ -67,6 +67,18 @@ module.exports = (eleventyConfig) => {
   });
 
   // Paired shortcodes...
+  eleventyConfig.addPairedLiquidShortcode(
+    'compatibility',
+    async (details, dart = null, libsass = null, ruby = null, feature = null) =>
```
Comment (mirisuzanne) [inline eleventy.config.js]: oh, this is a very clever solution, I need to use on my own projects.

File: eleventy.config.js [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+  eleventyConfig.addLiquidFilter('typogr', type.typogr);
+
+  // Dates
+  eleventyConfig.addLiquidFilter(
+    'formatDistanceToNow',
+    dates.formatDistanceToNow,
+  );
+
+  // Page
+  eleventyConfig.addLiquidFilter('isTypedoc', page.isTypedoc);
```
Comment (jgerigmeyer) [inline eleventy.config.js]: I'm not sure whether it's better to arrange these topically (like it is now), or by type of filter/shortcode/paired-shortcode. Thoughts?

File: eleventy.config.js [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+const syntaxHighlight = require('@11ty/eleventy-plugin-syntaxhighlight');
+const yaml = require('js-yaml');
+
+const codeExample = require('./source/helpers/codeExample.ts').default;
+const compatibility = require('./source/helpers/compatibility.ts');
+const components = require('./source/helpers/components.ts');
+const dates = require('./source/helpers/dates.ts');
+const { liquidEngine, markdownEngine } = require('./source/helpers/engines.ts');
+const page = require('./source/helpers/page.ts');
+const type = require('./source/helpers/type.ts');
```
Comment (jgerigmeyer) [inline eleventy.config.js]: I'm happy for feedback on how I organized and named these.

File: source/helpers/codeExample.ts [hunk 23]
```diff
@@ -0,0 +1,153 @@
...
+      splitLocation = 0.5 + ((maxSourceWidth - maxCSSWidth) / 110.0 / 2) * 100;
+    }
+  }
+
+  return {
+    scss: scssExamples,
+    sass: sassExamples,
+    css: cssExamples,
+    canSplit,
+    splitLocation,
```
Comment (jgerigmeyer) [inline source/helpers/codeExample.ts]: I'm not sure if this is going away with the change to CSS grid, or if that only impacts the padding/height of the blocks?

**Review by jgerigmeyer — COMMENTED** [review]
@mirisuzanne @stacyk This should be ready for style review/cleanup. ~~As far as I'm aware the missing piece of functionality is replacing the [former logic](https://github.com/oddbird/sass-site/blob/main/helpers/sass_helpers.rb#L168-L221) that added padding to ensure the various code blocks take up the same height. I think we're planning to replace that with CSS grid?~~

@sanajaved7 I did a round of JS organizing and documenting -- happy for your review/feedback on that!

**Comment by jgerigmeyer** [comment]
[Select Sass/SCSS Syntax](https://trello.com/c/NWEbNNgO/14-select-sass-scss-syntax)

File: eleventy.config.js (hunk 22 above)
Comment (mirisuzanne) [reply eleventy.config.js]: The other thing that we can do to clean it up a bit is have each individual file do an internal `module.exports = (eleventyConfig) => { ... }` with the added filters/codes, and then import them here as plugins. Don't know if that would feel a bit more organized?

Comment (jgerigmeyer) [reply eleventy.config.js]: @mirisuzanne Adjusted in https://github.com/oddbird/sass-site/commit/ad8a53c4a57a9df7efd8c0a06f1be05af34e86d5

File: eleventy.config.js (hunk 21 above)
Comment (jgerigmeyer) [reply eleventy.config.js]: Refactored in https://github.com/oddbird/sass-site/commit/ad8a53c4a57a9df7efd8c0a06f1be05af34e86d5 to do both... first by theme, then by type of filter.

**Review by jgerigmeyer — APPROVED** [review]
I added padding to the code examples in https://github.com/oddbird/sass-site/commit/31ba60a2cb7ef31f6525c7bfab529fce006e931b.

@mirisuzanne @stacyk @sanajaved7 I think this is ready for final review/merge! 🚀

File: source/helpers/components/index.ts [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+ * [Prism]: https://prismjs.com/
+ * [hl-tag]: https://www.11ty.dev/docs/plugins/syntaxhighlight/#usage
+ *
+ * @see https://prismjs.com/
+ */
+export const codeBlock = (contents: string, language: string, padding = 0) => {
+  if (!languages[language]) {
+    PrismLoader(language);
+  }
+  const code = `${contents}${'\n'.repeat(padding + 1)}`;
```
Comment (jgerigmeyer) [inline source/helpers/components/index.ts:36]: This padding isn't quite right when there are multiple stacked blocks... I'll take a look.

Comment (jgerigmeyer) [reply source/helpers/components/index.ts:36]: Oh, it's also slightly off in production: https://sass-lang.com/guide#topic-5

I'll take a look and see if I can find a quick fix...

File: source/assets/sass/components/_sass-syntax-switcher.scss [hunk 25]
```diff
@@ -110,7 +110,7 @@
   // syntax switcher is exactly equal to two lines, so there's no visual jitter
   // when switching between syntaxes.
   pre + pre {
-    margin-top: 22px;
+    margin-top: 1rem;
```
Comment (jgerigmeyer) [inline source/assets/sass/components/_sass-syntax-switcher.scss]: @mirisuzanne @stacyk This seems to fix the issue (also on the old site) with differing heights when multiple code blocks are stacked. See: https://github.com/oddbird/sass-site/pull/17/files#r1134278706

File: source/helpers/components/index.ts (hunk 24 above)
Comment (jgerigmeyer) [reply source/helpers/components/index.ts:36]: Fixed in https://github.com/oddbird/sass-site/commit/8d958ce4de17564aef0931f178f2c840a7336fd9

File: source/assets/sass/components/_sass-syntax-switcher.scss (hunk 25 above)
Comment (stacyk) [reply source/assets/sass/components/_sass-syntax-switcher.scss]: @jgerigmeyer are the tabs supposed to work in this branch?

File: source/_includes/code_examples/tab.liquid [hunk 26]
```diff
@@ -0,0 +1,3 @@
+<li class="ui-tabs-tab{% if syntax == 'css' %} css-tab{% endif %}{% if enabled %} ui-tabs-active{% endif %}">
```
Comment (stacyk) [inline source/_includes/code_examples/tab.liquid:1]: @jgerigmeyer  @sanajaved7  I am not sure if this is in scope of this PR, but I see there are classes on the live site that aren't included here (I think they are more of the jQuery classes like `ui-state-default` that controls the color of the tab link). I don't know if it makes sense to include that here, or IMO, that styling could be moved to the CSS instead of applied via js. 

I'm not sure if there was anything other than color that the missing classes were controlling.

That said, the tabs don't work locally so I'm wondering if I am missing something.

File: source/_includes/compatibility.liquid [hunk 27]
```diff
@@ -0,0 +1,35 @@
+<dl class="impl-status sl-c-description-list sl-c-description-list--horizontal">
+  <div class="compatibility">
+    Compatibility{% if feature %} ({{ feature }}){% endif %}:
+  </div>
+  {% if dart != null %}
+    <div>
+      <dt>Dart Sass</dt>
+      <dd>{{ dart | implStatus }}</dd>
```
Comment (stacyk) [inline source/_includes/compatibility.liquid:8]: I'm not sure how big a deal this is, but on the live site the version number (and not the word "since" is in an `<span class="caps">*</span>`

File: source/assets/sass/components/_sass-syntax-switcher.scss (hunk 25 above)
Comment (jgerigmeyer) [reply source/assets/sass/components/_sass-syntax-switcher.scss]: @stacyk Yes! They're working for me locally and on the deploy preview...

**Review by stacyk — COMMENTED** [review]
I'm wondering if this is why the nav wasn't shown on this page. If the page is a little wider, then we don't get these float issues on medium sizes screens. Is there a breakpoint declared to tell the page when to show/hide the CSS panel?
Screenshot 2023-03-13 at 4 33 57 PM

File: source/assets/sass/components/_sass-syntax-switcher.scss (hunk 25 above)
Comment (stacyk) [reply source/assets/sass/components/_sass-syntax-switcher.scss]: ok, it is working for me now... I will update my comments

File: source/_includes/code_examples/tab.liquid (hunk 26 above)
Comment (stacyk) [reply source/_includes/code_examples/tab.liquid:1]: Closing this. I shut it down and restarted it and now it is working as expected (the classes that are added and the styles)

**Comment by jgerigmeyer** [comment]
> I'm wondering if this is why the nav wasn't shown on this page. If the page is a little wider, then we don't get these float issues on medium sizes screens.

@stacyk Maybe? If we want more horizontal space, we can remove the nav, or raise the breakpoint for switching to one visible tab at a time.

> Is there a breakpoint declared to tell the page when to show/hide the CSS panel?

I think this is defined in [`_sass-syntax-switcher.scss`](https://github.com/oddbird/sass-site/blob/main/source/assets/sass/components/_sass-syntax-switcher.scss#L236-L246)? Splitting is disabled entirely for any examples where the longest scss/sass line and the longest css line add up to more than 110 chars.

File: source/_includes/compatibility.liquid (hunk 27 above)
Comment (jgerigmeyer) [reply source/_includes/compatibility.liquid:8]: Hm, I don't think this is a big deal, and from what I can tell neither of them are now in a `.caps` wrapper.

**Comment by stacyk** [comment]
> > Is there a breakpoint declared to tell the page when to show/hide the CSS panel?
> 
> I think this is defined in [`_sass-syntax-switcher.scss`](https://github.com/oddbird/sass-site/blob/main/source/assets/sass/components/_sass-syntax-switcher.scss#L236-L246)? Splitting is disabled entirely for any examples where the longest scss/sass line and the longest css line add up to more than 110 chars.


@jgerigmeyer I updated the breakpoints for this page and the documentation page (that one mostly just removing the function and using rems). I think this is looking good with these few changes. I will approve as I'm not sure the span caps thing should hold this up.

**Review by stacyk — APPROVED** [review]

File: source/_includes/compatibility.liquid (hunk 27 above)
Comment (stacyk) [reply source/_includes/compatibility.liquid:8]: Oh, is this just the homepage one? Im not sure why the homepage one doesn't use that caps span, but the code example pages do. 

It make such a small difference, I don't think we need to worry about it.

Comment (stacyk) [reply source/_includes/compatibility.liquid:8]: Screenshot 2023-03-13 at 9 42 31 PM

**Review by mirisuzanne — APPROVED** [review]
